pkg/apis/core/validation/validation.go

```diff
@@ -1930,16 +1930,18 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 			allErrs = append(allErrs, field.Required(fldPath.Child("dataSource", "name"), ""))
 		}
 
-		groupKind := schema.GroupKind{Group: "", Kind: spec.DataSource.Kind}
-		if spec.DataSource.APIGroup != nil {
-			groupKind.Group = string(*spec.DataSource.APIGroup)
-		}
-		groupKindList := make([]string, 0, len(supportedDataSourceAPIGroupKinds))
-		for grp := range supportedDataSourceAPIGroupKinds {
-			groupKindList = append(groupKindList, grp.String())
-		}
-		if !supportedDataSourceAPIGroupKinds[groupKind] {
-			allErrs = append(allErrs, field.NotSupported(fldPath.Child("dataSource"), groupKind.String(), groupKindList))
+		if !utilfeature.DefaultFeatureGate.Enabled(features.GenericPVCDataSource) {
```

Comment:
We can't add feature gate checking to validation in order to handle cases where someone persists an object with the new field, but then disables the feature later, or rolls back to a version with the feature disabled. Checking the feature gate in that case would cause updates/deletion to fail because the persisted object now contains invalid data.

Instead, we rely on `DropDisabledFields` to do the right thing before we get to validation.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#adding-unstable-features-to-stable-versions

I think the right thing to do here is to to remove the check for supported groupKinds, and only validate syntax.

Reply:
Thanks Michelle I understand what you mean